File: app/src/commands/report.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import { invoke } from "@tauri-apps/api/core";
import type { ForwardReportRequest, PlayerId, ReportKind } from "@tsukilabs/nil-bindings";

export async function forwardReport(recipient: PlayerId, report: ReportKind) {
  const req: ForwardReportRequest = {
    world: NIL.world.getIdStrict(),
    recipient,
    report,
  };

  await invoke("forward_report", { req });
}

File: crates/nil-core/src/world/report.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use crate::error::Result;
use crate::player::PlayerId;
use crate::report::ReportKind;
use crate::world::World;

impl World {
  /// Forwards a report to a player.
  pub fn forward_report(&self, recipient: PlayerId, report: ReportKind) -> Result<()> {
    if self.has_player(&recipient) {
      self.emit_report(recipient, report)?;
    }

    Ok(())
  }
}
